### PR TITLE
Create Customizable Toolbar

### PR DESCRIPTION
**Pull request by cooltey**
https://phabricator.wikimedia.org/T296753

### CODE REVIEW

**Comment by cooltey** [comment]
@dbrant @sharvaniharan The PR is ready for review.

**Review by sharvaniharan — CHANGES_REQUESTED** [review]
Looks very good and is very seamless! 
One minor change.. the icon tint looks off from what i can see on the ticket..

**Comment by sharvaniharan** [comment]
One other thing, please review and merge this: #3040 and wire-it up on exit of this customization screen.

**Comment by cooltey** [comment]
> Looks very good and is very seamless! One minor change.. the icon tint looks off from what i can see on the ticket..

Done!

File: app/src/main/java/org/wikipedia/page/customize/CustomizeQuickActionsActivity.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import org.wikipedia.analytics.SuggestedEditsFunnel
+
+class CustomizeQuickActionsActivity : SingleFragmentActivity<CustomizeQuickActionsFragment>() {
+    public override fun createFragment(): CustomizeQuickActionsFragment {
+        return CustomizeQuickActionsFragment.newInstance()
+    }
+
+    companion object {
+        fun newIntent(context: Context): Intent {
+            SuggestedEditsFunnel.get().contributionsOpened()
```
Comment (sharvaniharan) [inline app/src/main/java/org/wikipedia/page/customize/CustomizeQuickActionsActivity.kt:15]: Please remove this @cooltey

Comment (cooltey) [reply app/src/main/java/org/wikipedia/page/customize/CustomizeQuickActionsActivity.kt]: Done

**Comment by cooltey** [comment]
@sharvaniharan @dbrant This PR passed the design review and is ready for code review now.

File: app/src/main/java/org/wikipedia/theme/ThemeChooserDialog.kt [hunk 2]
```diff
@@ -86,7 +86,12 @@ class ThemeChooserDialog : ExtendedBottomSheetDialogFragment() {
         updateComponents()
         disableBackgroundDim()
         requireDialog().window?.let {
-            DeviceUtil.setNavigationBarColor(it, getThemedColor(requireContext(), R.attr.paper_color))
+            DeviceUtil.setNavigationBarColor(it, ResourceUtil.getThemedColor(requireContext(), R.attr.paper_color))
+        }
+
+        // TODO: test only
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/theme/ThemeChooserDialog.kt:92]: Is this still a TODO?

**Review by dbrant — CHANGES_REQUESTED** [review]
Functionally this works very well! Just a few requests and observations:

* When rearranging the menu items in the list, it looks like grabbing anywhere in an item causes it to start rearranging. This makes it impossible to _scroll_ the whole list up and down. Can we make it so that the items start rearranging only when grabbing the rearrange button on the right? (similar to how the Language list works)

* Have we thought about how we should handle long captions on toolbar buttons that overflow onto multiple lines?

image

* I'm not a designer, but the visual appearance of the "Customize toolbar" button clashes with almost every other UI element in the app. The color doesn't match anything in the WMF guidelines. Are we sure about this?

image

File: app/src/main/java/org/wikipedia/theme/ThemeChooserDialog.kt (hunk 2 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/theme/ThemeChooserDialog.kt:92]: Removed in my PR @dbrant https://github.com/wikimedia/apps-android-wikipedia/pull/3060/files#diff-c1eae10dbe3c866e252b9d1e4a1386098e0313ee4b3e23c11f23d02360b2ab59L92

**Comment by cooltey** [comment]
> Functionally this works very well! Just a few requests and observations:
> 
>     * When rearranging the menu items in the list, it looks like grabbing anywhere in an item causes it to start rearranging. This makes it impossible to _scroll_ the whole list up and down. Can we make it so that the items start rearranging only when grabbing the rearrange button on the right? (similar to how the Language list works)

Yes, I think this also makes sense to me. Updated! 

And I'll leave a comment on the ticket.

>     * Have we thought about how we should handle long captions on toolbar buttons that overflow onto multiple lines?
> 

This too, I'll let Robin and Jazmin know about it.

**Comment by cooltey** [comment]
> Functionally this works very well! Just a few requests and observations:
> 
>     * When rearranging the menu items in the list, it looks like grabbing anywhere in an item causes it to start rearranging. This makes it impossible to _scroll_ the whole list up and down. Can we make it so that the items start rearranging only when grabbing the rearrange button on the right? (similar to how the Language list works)
> 

This one is resolved. Please let me know how it looks to you.

**Pull request merged**